File: web/application/models/Campo.php
<?php

class Campo {
	private $id;
	private $titulo;
	private $maxlength;
	private $regex;
	private $tipo;
	private $status;
	private $data;
	private $id_categoria = array();

	public function getId() {
	    return $this->id;
	}

	public function setId($id) {
	    return $this->id = $id;
	}

	public function getTitulo() {
	    return $this->titulo;
	}

	public function setTitulo($titulo) {
	    return $this->titulo = $titulo;
	}

	public function getMaxlength() {
	    return $this->maxlength;
	}

	public function setMaxlength($maxlength) {
	    return $this->maxlength = $maxlength;
	}

	public function getRegex() {
	    return $this->regex;
	}

	public function setRegex($regex) {
	    return $this->regex = $regex;
	}

	public function getTipo() {
	    return $this->tipo;
	}

	public function setTipo($tipo) {
	    return $this->tipo = $tipo;
	}

	public function getStatus() {
	    return $this->status;
	}

	public function setStatus($status) {
	    return $this->status = $status;
	}

	public function getData() {
	    return $this->data;
	}

	public function setData($data) {
	    return $this->data = $data;
	}

	public function getIdCategoria() {
	    return $this->id_categoria;
	}

	public function setIdCategoria($id_categoria) {
	    return $this->id_categoria = $id_categoria;
	}
}

File: trunk/java/backup projeto antigo/src/com/ifsp/integrado/model/Campo.java
package com.ifsp.integrado.model;

import java.util.ArrayList;
import java.util.Date;

public class Campo extends Model {
	
	private int id;
	private String titulo;
	private int maxlength;
	private String regex;
	private char tipo;
	private char status;
	private Date data;
	private ArrayList<Categoria> id_categoria;

	public int getId() {
		return id;
	}
	
	public void setId(int id) {
		this.id = id;
	}
	
	public String getTitulo() {
		return titulo;
	}
	
	public void setTitulo(String titulo) {
		this.titulo = titulo;
	}
	
	public int getMaxlength() {
		return maxlength;
	}
	
	public void setMaxlength(int maxlength) {
		this.maxlength = maxlength;
	}
	
	public String getRegex() {
		return regex;
	}
	
	public void setRegex(String regex) {
		this.regex = regex;
	}
	
	public char getTipo() {
		return tipo;
	}
	
	public void setTipo(char tipo) {
		this.tipo = tipo;
	}
	
	public char getStatus() {
		return status;
	}
	
	public void setStatus(char status) {
		this.status = status;
	}
	
	public Date getData() {
		return data;
	}
	
	public void setData(Date data) {
		this.data = data;
	}

	public void setIdCategoria(Categoria categoria) {
		this.id_categoria.add(categoria);
	}
	
	public ArrayList<Categoria> getIdCategoria() {
		return id_categoria;
	}

	@Override
	public void validate() {
		// TODO Auto-generated method stub
		
	}

	/**
	 * 
	 * @param Campo campo Novo objeto após assemblyRequest()
	 * @return
	 */
	public Campo mergeObjects(Campo campo) {
		int id        = campo.getId();
		String titulo = campo.getTitulo();
		int maxlength = campo.getMaxlength();
		String regex  = campo.getRegex();
		char tipo     = campo.getTipo();
		char status   = campo.getStatus();
		Date data     = campo.getData();
		
		if (this.isNotNullOrEmpty(String.valueOf(id)) && id != 0) {
			this.id = id;
		}
		
		if (this.isNotNullOrEmpty(titulo)) {
			this.titulo = titulo;
		}
		
		if (this.isNotNullOrEmpty(String.valueOf(maxlength)) && maxlength != 0) {
			this.maxlength = maxlength;
		}
		
		if (this.isNotNullOrEmpty(regex)) {
			this.regex = regex;
		}
		
		if (this.isNotNullOrEmpty(String.valueOf(tipo))) {
			this.tipo = tipo;
		}
		
		if (this.isNotNullOrEmpty(String.valueOf(status))) {
			this.status = status;
		}
		
		if (this.isNotNullOrEmpty(data.toString())) {
			this.data = data;
		}
		
		return campo;
	}
		
}
